uberenv.py

```diff
@@ -493,32 +522,51 @@ def install(self):
         # note: this assumes package extends python when +python
         # this may fail general cases
         if self.opts["install"] and "+python" in full_spec:
-            activate_cmd = "spack/bin/spack activate " + self.pkg_name
+            activate_cmd = "spack/bin/spack activate " + self.pkg_name + self.opts["spec"]
```

Comment:
Could use hash here: package already installed at this point, so hash is the robust way.

Reply:
Done